tests/vm_test.go

```diff
@@ -1891,10 +1886,13 @@ var _ = Describe("[rfe_id:1177][crit:medium][vendor:cnv-qe@redhat.com][level:com
 				Expect(vmRunningRe.FindString(stdout)).ToNot(Equal(""), "VMI is not Running")
 
 				By("Checking no DeletionTimestamp was set")
-				newVMI, err := virtClient.VirtualMachineInstance(vmi.Namespace).Get(vmi.Name, &k8smetav1.GetOptions{})
-				Expect(err).ToNot(HaveOccurred())
-				Expect(newVMI.ObjectMeta.DeletionTimestamp).To(BeNil())
-				Expect(reflect.DeepEqual(newVMI, originalVMI)).To(BeTrue())
+				Consistently(func() error {
+					actualVMI, err := virtClient.VirtualMachineInstance(vmi.Namespace).Get(vmi.Name, &k8smetav1.GetOptions{})
+					Expect(err).ToNot(HaveOccurred())
+					Expect(actualVMI.ObjectMeta.DeletionTimestamp).To(BeNil())
+					return nil
+				}, 30*time.Second, 1*time.Second).Should(BeNil())
```

Comment:
Waiting 30sec and running the fetch every 1sec is aggressive. Can this be soften, e.g. check every 5sec?
But I do not understand how this solved any flakiness. Maybe the claim is that the current test is not really checking well, because it may take more time to see it failing.

Hope that the commit message will clarify this.

Reply:
Thanks @EdDev I will change commit and PR descriptions to be more clear. Just a preview:
The problem was that there was a `DeepEqual` between two retrieved VMIs (one before the `stop --dry-run` command and one after). This assertion was too strong because there are several fields that can change independently from the command. IMO, since `stop` command causes the deletion of the vmi, checking that the `DeletionTimestamp` remains `nil` is enough. What do you think? I will also change the fetching intervals. Thanks again :slightly_smiling_face:

Reply:
Thank you for the clarification.

Does the `stop` command performs a post to change the VMI CR?

Reply:
At `virt-api` level the `stop` command performs a patch on the VMI CR.

Reply:
Great, so can't you just check that this patch occurred? I am guessing that `DeletionTimestamp` is only a side effect of the controllers acting upon the request. But the dry-run is not suppose to touch the manifest on the api server eventually, so this check sounds much faster to me.

Reply:
AFAIK, `dry-run` operations means that the CR is not to be touched on the API server.
It will pass through the admission-control, but should not be persisted. So just checking that the CR was not updated with the "stop" instruction (I do not know what it changes), then you are suppose to cover this.

Reply:
Correct. Checking for a change of `Running`, `RunStrategy` or absence of `StopRequest` will be enough.

Reply:
Thanks so much for the clarification. Now I got it.

Reply:
Per @xpivarc confirmation, we better now check against the CR and this probably does not require waiting at all.